tee-worker/cli/src/base_cli/commands/litentry/link_identity.rs

```diff
@@ -55,7 +55,8 @@ impl LinkIdentityCommand {
 		let nonce = direct_api.get_next_nonce(&shard, &self.get_primary_account_id()).unwrap();
 
 		let signer = self.get_signer();
-		chain_api.set_signer(signer.clone().into());
+		let who: sr25519_core::Pair = get_pair_from_str(&self.account).into();
```

Comment:
can you educate me how this line brings more value compared to before?

Reply:
The single line itself doesn't bring more value, but the changes in the file collectively alter the logic. So earlier we were using `let signer = self.get_signer()` and then using it to set the signer of the extrinsic and also in the extrinsic parameter. However, when using a delegate signer, we are setting the delegate signer as the parameter in the extrinsic. So to avoid this, I'm creating a separate `who` for the primary account, which will go to the `extrinsic` parameter and `signer` will be the signer of the extrinsic depending on what arg/flag has been provided.

Reply:
got it, it makes sense now, thanks!